src/agent/tests/register_test.cpp

```diff
@@ -1,12 +1,57 @@
 #include <register.hpp>
 
+#include <agent_info.hpp>
+
+#include <gmock/gmock.h>
 #include <gtest/gtest.h>
 
-TEST(RegistrationTest, RegistrationTest)
+using ::testing::_;
+using ::testing::MockFunction;
+using ::testing::Return;
+
+class RegisterTest : public ::testing::Test
+{
+protected:
+    const AgentInfo* agent;
+
+    MockFunction<std::optional<std::string>(
+        const std::string&, const std::string&, const std::string&, const std::string&)>
+        MockAuthenticateFunction;
+    MockFunction<http::status(const std::string& host,
+                              const std::string& port,
+                              const std::string& token,
+                              const std::string& uuid,
+                              const std::string& key,
+                              const std::optional<std::string>& name)>
+        MockRegistrationFunction;
+
+    ~RegisterTest()
+    {
+        delete agent;
+    }
+    void SetUp()
+    {
+        agent = new AgentInfo("agent_name", "agent_key", "fbdea7a6-0e0d-4d2c-8af3-9330ea8799a3");
+    }
+    void TearDown()
+    {
+        delete agent;
```

Comment:
Better to use reset with unique ptr https://en.cppreference.com/w/cpp/memory/unique_ptr/reset

Reply:
Actually, since we are creating a new agent on `SetUp`, there's no need to call `reset`. It will happen automatically on assignment.